packages/grafana-ui/src/components/Combobox/useLatestAsyncCall.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useRef } from 'react';
+
+type AsyncFn<T, V> = (value: T) => Promise<V>;
+
+export function useLatestAsyncCall<T, V>(fn: null | AsyncFn<T, V>): AsyncFn<T, V> {
+  const latestValue = useRef<T>();
+
+  const wrappedFn = useCallback(
+    (value: T) => {
+      if (!fn) {
+        throw new Error('useLatestAsyncCall was called with a null function');
+      }
+
+      latestValue.current = value;
+
+      return new Promise<V>((resolve, reject) => {
+        fn(value).then((result) => {
+          if (latestValue.current === value) {
```

Comment:
Maybe add a comment to describe why this works here

Reply:
Sort of nitpicky but could make this be a bit unpredictable is that if value is what you type, you can go back and forth between values. So older value can be the same as newer one. If the response from the API is stable in time this is ok, but for example if it takes more time and over that time something changes in the backend, we can still accept the old response maybe missing some options that were added in between.

I think it would be better to just generate a unique token here instead of using actual value to see whether the response is stale.

Reply:
agree, but instead of generating a token i think it should just be timestamp based. just grab `Date.now()` when the request is made, and only update the state if the timestamp is later than the newest

see e.g. https://github.com/grafana/grafana/pull/68377/files

Reply:
@ashharrison90 Yeah, I beleive that could be a better approach. I tried a few attempts at using timestamp briefly and couldn't get it to work, so I've punted this for now.